src/Visualization/Python/PlotMemoryMonitors.py

```diff
@@ -0,0 +1,269 @@
+                    # the values in 'df'.
+                    total = df[column].copy()
+
+        return total
+
+    # Get a list of all components that we have monitored from the first
+    # reductions file
+    memory_filenames = check_memory_monitor_dir(reduction_files[0])
+
+    # Open every h5file. For each h5file, turn each subfile into a DataFrame.
+    # Then concat all DataFrames together into one that's indexed by the
+    # subfile/component name.
+    totals_df = pd.DataFrame()
+    for reduction_file in reduction_files:
+        h5file = h5py.File(reduction_file)
+        local_totals_df = pd.DataFrame()
+
+        for subfile_name in memory_filenames:
+            df = check_memory_monitor_file(reduction_file, h5file, subfile_name)
+            if "Time" not in local_totals_df:
+                local_totals_df["Time"] = df["Time"]
+            local_totals_df[subfile_name] = total_over_nodes(df)
+
+        totals_df = pd.concat([totals_df, local_totals_df])
+
+    # Restrict to x-bounds if there are any
+    for subfile in memory_filenames:
+        if x_bounds is not None:
+            totals_df[subfile] = totals_df[subfile][
+                (totals_df["Time"] >= x_bounds[0])
+                & (totals_df["Time"] <= x_bounds[1])
+            ]
+
+    # Need .dat because all other components have that extension
+    the_rest_str = "The Rest.dat"
+
+    # Given a dictionary of DataFrames, sum all the columns from all the
+    # DataFrames that list totals for each node to get a grand total memory
+    # usage. Then return a DataFrame with the time and this new total
+    def compute_overall_total_and_the_rest(df: pd.DataFrame):
+        total = None
+        for component_name in memory_filenames:
+            if total is not None:
+                total += df[component_name]
+            else:
+                total = df[component_name].copy()
+
+        average_max = total.mean()
+        # Always plot total
+        components_to_plot = []
+
+        # If the max of a component is too small (less than 1% here) just
+        # group them all together in a category called "The Rest". Otherwise
+        # keep the original component
+        the_rest = None
+        for component_name in memory_filenames:
+            component_total = df[component_name]
+            if component_total.max() < 0.01 * average_max:
+                if the_rest is not None:
+                    the_rest += component_total
+                else:
+                    the_rest = component_total.copy()
+            else:
+                components_to_plot.append(component_name)
+
+        if the_rest is not None:
+            components_to_plot.append(the_rest_str)
+
+        return total, the_rest, components_to_plot
+
+    # Now add the total and the rest to the totals DataFrame
+    total_series, the_rest_series, components_to_plot = (
+        compute_overall_total_and_the_rest(totals_df)
```

Comment:
You do this only once, so you don't need the function I think? Also update the comment about "dicts" above.